src/main/java/org/codehaus/mojo/exec/ExecMojo.java

```diff
@@ -280,7 +290,7 @@ else if ( !StringUtils.isEmpty( argsProp ) )
 
             commandLine.addArguments( args, false );
 
-            Executor exec = new DefaultExecutor();
+            Executor exec = inheritIo ? new ExtendedExecutor(inheritIo) : new DefaultExecutor();
```

Comment:
I don't understand the constructor; This will always be `new ExtendedExecutor(true)`

Reply:
That was a way to minimize code changes, but a better fix would be
```
Executor exec = new ExtendedExecutor(inheritIo);
```